codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/shapevisitor/DafnyToSmithyShapeVisitor.java

```diff
@@ -211,6 +241,15 @@ public String blobShape(final BlobShape shape) {
 
   @Override
   public String structureShape(final StructureShape shape) {
+    if (SmithyNameResolver.isShapeFromAWSSDK(shape)) {
+      writer.addImportFromModule(
+        SmithyNameResolver.getGoModuleNameForSdkNamespace(
+          shape.getId().getNamespace()
+        ),
+        "types",
+        SmithyNameResolver.smithyTypesNamespace(shape)
+      );
+    }
```

Comment:
Same, all this can be moved inside the NameResolver and we can then avoid this repetition.

Reply:
Let's do it in next PR once positional is merged.